docker-compose.yml

```diff
@@ -0,0 +1,31 @@
+services:
+  db:
+    image: postgres:17-alpine
+    restart: always
+    ports:
+      - "5432:5432"
+    env_file:
+      - .env
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+
+  app:
+    build: .
+    command: >
+      sh -c "python manage.py wait_for_db &&
+            python manage.py makemigrations &&
+            python manage.py migrate &&
+            python manage.py runserver 0.0.0.0:8000"
+    volumes:
+      - ./:/app
+      - media_volume:/app/media
+    ports:
+      - "8000:8000"
+    env_file:
+      - .env
+    depends_on:
+      - db
```

Comment:
The `depends_on` keyword does not wait for the database to be fully ready; it only ensures that the container is started. Consider using a script like `wait-for-it` or `wait-for-db` to ensure the database is ready before the Django app starts.